zulip_bots/zulip_bots/bots/ping/ping.py

```diff
@@ -0,0 +1,117 @@
+import requests
+import time
+from typing import Any, Dict, Optional, Text
+
+class PingHandler(object):
+    PING_API = "https://check-host.net/check-ping"
+    RESULT_API = "https://check-host.net/check-result/{request_id}"
+
+    INITIAL_MESSAGE = "Checking availability of `{host}`, please wait..."
+    EMPTY_HOST_MESSAGE = "The hostname is empty."
+    INVALID_HOST_MESSAGE = "The hostname `{host}` is invalid."
+
+    NODE_DATA = "{country}, {city} ({ip_address})"
+    NODE_RESPONSES = {
+        'UNKNOWN': ":cross_mark: {node_data} – Unknown host",
+        'OK': ":heavy_check_mark: {node_data}",
+        'MALFORMED': ":cross_mark: {node_data} – Malformed reply",
+        'TIMEOUT': ":cross_mark: {node_data} – Timed out"
+    }
+    RESPONSE_MESSAGE = "Ping results for `{host}`:\n\n" \
+                       "{node_responses}\n\n" \
+                       "You can see the full report here: {link}."
+
+    def usage(self) -> Text:
+        return '''
+        This bot allows users to check availability of a given website
+        or IP address across 10 random locations.
+        '''
+
+    def ping(self, host: Text) -> Any:
+        response = requests.get(url=self.PING_API,
+                                params={'host': host, 'max_nodes': '10'},
+                                headers={'accept': 'application/json'})
+        if response.status_code != 200:
+            response.raise_for_status()
```

Comment:
Can you catch the exception somewhere? (If we make this an embedded bot, throwing an uncaught exception would mean an email sent to the administrator.)